### ISSUE TITLE
[YCQL]  failed create index is not dropped; trying to drop it later crashes yb-master

### ISSUE THREAD

**Issue by kmuthukk** [yugabyte/yugabyte-db#5139]
The test creates a simple table where the name column has some duplicates. It then attempts to create a unique index on the name column-- and this operation is expected to fail since the table contains duplicate entries for the name column.

But 'describe' on the table still shows the index is present. The good part is that it seems to be in a failed state internally - so the query plan doesn't use the index. An attempt to later explicity drop the index crashes the yb-master.

```
ycqlsh> CREATE KEYSPACE k;
ycqlsh> USE k;
ycqlsh:k> CREATE TABLE IF NOT EXISTS users(id text, name text, age int, PRIMARY KEY(id)) WITH TRANSACTIONS = {'enabled' : true};
ycqlsh:k> INSERT INTO users(id, name, age) VALUES ('1', 'abc', 10);
ycqlsh:k> INSERT INTO users(id, name, age) VALUES ('2', 'abc', 10);
ycqlsh:k> select * from users;

 id | name | age
----+------+-----
  2 |  abc |  10
  1 |  abc |  10

(2 rows)
ycqlsh:k> CREATE UNIQUE INDEX IF NOT EXISTS name_idx ON users(name);
ycqlsh:k> DESCRIBE users;

CREATE TABLE k.users (
    id text PRIMARY KEY,
    name text,
    age int
) WITH default_time_to_live = 0
    AND transactions = {'enabled': 'true'};
CREATE UNIQUE INDEX name_idx ON k.users ()
    WITH transactions = {'enabled': 'true'};

-- As expected, the failed index is NOT used by the query planner.
ycqlsh:k> EXPLAIN SELECT * FROM users WHERE name='abc';

 QUERY PLAN
--------------------------
 Seq Scan on k.users
   Filter: (name = 'abc')

ycqlsh:k> DROP INDEX name_idx;
OperationTimedOut: errors={'127.0.0.1': 'Client request timeout. See Session.execute[_async](timeout)'}, last_host=127.0.0.1
```

But an attempt to drop the "failed" index crashes yb-master, with the following FATAL:

```
F20200718 04:12:21 ../../src/yb/master/catalog_manager.cc:3894] Check failed: l->data().pb.state() == SysTablesEntryPB::ALTERING
    @     0x7f220d2d392c  yb::LogFatalHandlerSink::send()
    @     0x7f220c4b9376  google::LogMessage::SendToLog()
    @     0x7f220c4b67da  google::LogMessage::Flush()
    @     0x7f220c4b98a9  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f2217a3e81f  yb::master::CatalogManager::GetTableSchema()
    @     0x7f2217ab8e92  yb::master::MasterServiceImpl::GetTableSchema()
    @     0x7f2210abcbd0  yb::master::MasterServiceIf::Handle()
    @     0x7f220eb091d9  yb::rpc::ServicePoolImpl::Handle()
    @     0x7f220eaac074  yb::rpc::InboundCall::InboundCallTask::Run()
    @     0x7f220eb14f28  yb::rpc::(anonymous namespace)::Worker::Execute()
    @     0x7f220d368f3f  yb::Thread::SuperviseThread()
    @     0x7f2208a98694  start_thread
    @     0x7f22081d541d  __clone
```

**Comment by tedyu** [comment]
Looks like the following in MultiStageAlterTable::UpdateIndexPermission wasn't executed :
```
      indexed_table_data.set_state(SysTablesEntryPB::ALTERING,
```
Maybe CatalogManager can use the state not being SysTablesEntryPB::ALTERING to infer that index was not successfully created.

Another choice is to remove index metadata from the underlying table if the index cannot be created.

**Comment by amitanandaiyer** [comment]
Dropping the index will be done as part of #5144

**Comment by kmuthukk** [comment]
@amitanandaiyer -- let's backport this to 2.2.x line also. Thx.

**Comment by amitanandaiyer** [comment]
@kmuthukk This is already landed into 2.2.0

did you intend to say that #5144 should be backported? #5144 is atm only in the master.

**Comment by kmuthukk** [comment]
yes - sorry, would be good backport 5144 to 2.2.0 also.
